src/dataworker/Dataworker.ts

```diff
@@ -580,40 +622,91 @@ export class Dataworker {
     await Promise.all(Object.values(spokePoolClients).map((client: SpokePoolClient) => client.update()));
 
     // TODO:
-    // 3. Create roots.
+    // 2. Create roots
     const poolRebalanceRoot = this.buildPoolRebalanceRoot(blockRangesForProposal, spokePoolClients);
-    console.log(`poolRebalanceRoot:`, poolRebalanceRoot.tree.getHexRoot());
-    poolRebalanceRoot.leaves.forEach((leaf: PoolRebalanceLeaf) => {
+    poolRebalanceRoot.leaves.forEach((leaf: PoolRebalanceLeaf, index) => {
       const prettyLeaf = Object.keys(leaf).reduce((result, key) => {
         // Check if leaf value is list of BN's. For this leaf, there are no BN's not in lists.
         if (BigNumber.isBigNumber(leaf[key][0])) result[key] = leaf[key].map((val) => val.toString());
         else result[key] = leaf[key];
         return result;
       }, {});
-      console.log(prettyLeaf);
+      this.logger.debug({
+        at: "Dataworker",
+        message: `Pool rebalance leaf #${index}`,
+        leaf: prettyLeaf,
```

Comment:
this might let the log be done on one line. best to try do this when the file is already over 600 lines long...
```suggestion
        prettyLeaf,
```